pandas/io/formats/style_render.py

```diff
@@ -485,19 +483,38 @@ def format(
         if not isinstance(formatter, dict):
             formatter = {col: formatter for col in columns}
 
-        for col in columns:
-            format_func = _maybe_wrap_formatter(
-                formatter.get(col),
-                na_rep=na_rep,
-                precision=precision,
-                decimal=decimal,
-                thousands=thousands,
-                escape=escape,
-            )
+        if self.index.is_unique and self.columns.is_unique:
+            for col in columns:
+                format_func = _maybe_wrap_formatter(
+                    formatter.get(col),
+                    na_rep=na_rep,
+                    precision=precision,
+                    decimal=decimal,
+                    thousands=thousands,
+                    escape=escape,
+                )
+
+                j = self.columns.get_loc(col)  # single value
+                for row, value in data[[col]].itertuples():
+                    i = self.index.get_loc(row)  # single value
```

Comment:
if you pull the col indexer out does this still have a perf hit? e.g. get_indexer_for calls get_loc if its unique (which is cached) anyhow.

Reply:
Good reviewing! Third time's a charm, the code is simpler, works for non-unique and unique and performance improvement:

```
       before           after         ratio
     [4af3eed5]       [c3b7af82]
     <text_gradient^2>       <styler_non_unique>
-      33.1±0.4ms       25.9±0.5ms     0.78  io.style.Render.time_format_render(12, 120)
-      86.4±0.4ms       66.8±0.5ms     0.77  io.style.Render.time_format_render(36, 120)
-        62.3±2ms       46.4±0.4ms     0.75  io.style.Render.time_format_render(24, 120)
-      10.1±0.3ms       4.16±0.2ms     0.41  io.style.Render.time_format_render(12, 12)
-      24.7±0.2ms       9.53±0.2ms     0.39  io.style.Render.time_format_render(36, 12)
-      17.7±0.2ms       6.76±0.1ms     0.38  io.style.Render.time_format_render(24, 12)

SOME BENCHMARKS HAVE CHANGED SIGNIFICANTLY.
PERFORMANCE INCREASED.
```